### DISCUSSION TITLE
Would it make sense to have a directory lookup to discover interesting commands?

### DISCUSSION THREAD

**Discussion by mofosyne** [tldr-pages/tldr#15347] [Ideas]
Much like how books has an index page that may be organize in categories... or the start menu of windows has subcategories of different programs... maybe we want something like that as well?

Alternatively we could adopt a hashtag like system and have a standardized hierarchy of hashtag for hashtag discovery.

If this doesn't make sense to be in this project, then let me know if it would make sense for a 'start button' like cli app for app/cli discovery.

**Comment by sbrl** [comment]
Not sure I understand what you're saying here, @mofosyne.

tldr-pages already organises pages by platform in the current directory structure.

Can you elaborate further?

Note that tldr-pages/tldr is generally about only the pages thsemlves. However, anyone is free to create their own client, and we have a large ecosystem of clients that have been written by the community: https://github.com/tldr-pages/tldr/wiki/Clients

....so feel free to implement a new client (or fork an existing one!) and add the new feature you suggest yourself.

We have a client spec here as guidance to help you along: <https://github.com/tldr-pages/tldr/blob/main/CLIENT-SPECIFICATION.md>

**Comment by mofosyne** [comment]
Essentially, the idea would be to allow each pages to be organized into categories and subcategories. 

For example many desktop start menu often automatically organize programs into groups for easier discoverability (in fact, you would notice that they often combine the typical 'search box' with the grouped program selection... in which the search box would be equivalent to `tldr <page name>`) (e.g.  'Design Tool > Electronic Design Automation > KiCAD')

How this would be done is something I'm speculating. I imagine this can either come in the form of:
* A separate database
    * Pros: No synchronization needed between page and this category metadata
    * Cons: Can become desynced if the page changes context
* Marked up in the actual page themselves
    * Pros: Page authors would be encouraged to consider what this fits into
    * Cons: Will need to have some machine readable syntax

This could be useful for non program based discovery, e.g. looking up for 'linux standard folder structure' reference which is not a program.
